server/src/services/cart.service.ts

```diff
@@ -39,6 +40,52 @@ class CartService {
 
     return carts;
   }
+
+  public async updateCart(
+    transaction: Transaction,
+    user_id: string,
+    cart: Pick<ICartData, 'id' | 'option_id' | 'amount'>
+  ): Promise<void> {
+    // check validation
+    const currentCart = await Cart.findByPk(cart.id);
+    if (!currentCart) {
+      throw new CartError(CartErrorType.NOT_VALID_CART_ID);
+    }
+    await this._checkOptionValid(currentCart.item_id, cart.option_id);
+
+    const [updateSuccess] = await Cart.update(
+      {
+        option_id: cart.option_id,
+        amount: cart.amount,
+      },
+      {
+        where: {
+          id: cart.id,
+          user_id,
+        },
+        transaction,
+      }
+    );
+
+    if (!updateSuccess) {
+      // cart id 가 유저가 소유한 cartId가 아닐때 발생
+      throw new CartError(CartErrorType.NOT_VALID_CART_ID);
+    }
+  }
+
+  public async deleteCart(user_id: string, id: number) {
+    const deleteCount = await Cart.destroy({
+      where: {
+        id,
+        user_id,
+      },
+    });
+
+    if (deleteCount >= 1) {
+      return;
+    }
+    throw new CartError(CartErrorType.NOT_VALID_CART_ID);
```

Comment:
```suggestion
    if (deleteCount === 0) {
        throw new CartError(CartErrorType.NOT_VALID_CART_ID);
    }
```

else일 때 throw를 하는 것보단,
특정 상황일 때 에러를 발생시키는 편이 낫지 않나 싶습니다! 👀